### ISSUE TITLE
0.20 incompatibility with text module?

### ISSUE THREAD

**Issue by rpkilby** [systemjs/systemjs#1550]
This is a bit beyond me - trying to provide as much information as possible. 

The `define` function in `format-helpers` seems to be failing with the `text` plugin. With the following config:
```js
SystemJS.config({
  ...
  meta: {
    "*.html": {
      "loader": "text"
    }
  }
});
```

Loading an html template will result in these arguments passed to `define`:

#### name
```js
function () {
    return "\n<div class=\"panel panel-default\">…"  // more html
}
```

#### deps
```js
undefined
```

and where `curMetaDeps` is an empty array. 

Since an empty array is truthy, this [block](https://github.com/systemjs/systemjs/blob/0.20.0/src/format-helpers.js#L57-L60) is entered and attempts to call the `concat` method on a function. 

I was able to patch this locally with an "is empty" check, replacing 

```js
if (curMetaDeps) {
```
with 
```js
if (curMetaDeps && Object.keys(curMetaDeps).length !== 0) {
```

While this works, I'm not sure if the check is correct/appropriate.

**Comment by guybedford** [comment]
Thanks so much for posting. I've released a fix in 0.20.1.

**Comment by rpkilby** [comment]
The patch fixes the issue. Many thanks. 👍  👍  👍 

Also, made my day. Patch was posted before I could even finish downgrading.
